OrbitCore/LinuxEventTracer.cpp

```diff
@@ -115,27 +119,29 @@ void LinuxEventTracerThread::Run(
     }
 
     new_events = false;
-    pid_t new_thread = -1;
+    pid_t thread_forked = -1;
+    pid_t thread_exited = -1;
 
     // Read from all ring buffers, create events and store them in the
     // event_queue. In order to ensure that no buffer is read constantly while
     // others overflow, we schedule the reading using round-robin like
     // scheduling.
-    for (LinuxPerfRingBuffer& ring_buffer : ring_buffers) {
-      if (*a_ExitRequested || new_thread >= 0) {
+    for (auto& fd_to_ring_buffer : fds_to_ring_buffer) {
+      if (*a_ExitRequested || thread_forked >= 0 || thread_exited >= 0) {
         break;
       }
 
-      bool is_uprobes =
-          uprobe_fds_to_function.count(ring_buffer.FileDescriptor()) > 0;
-      bool is_uretprobes =
-          uretprobe_fds_to_function.count(ring_buffer.FileDescriptor()) > 0;
+      const int32_t& fd = fd_to_ring_buffer.first;
+      LinuxPerfRingBuffer& ring_buffer = fd_to_ring_buffer.second;
+
+      bool is_uprobes = uprobe_fds_to_function.count(fd) > 0;
+      bool is_uretprobes = uretprobe_fds_to_function.count(fd) > 0;
 
       uint32_t read_from_this_buffer = 0;
       // Read up to ROUND_ROBIN_BATCH_SIZE (5) new events
       while (ring_buffer.HasNewData() &&
              read_from_this_buffer < ROUND_ROBIN_BATCH_SIZE) {
-        if (*a_ExitRequested || new_thread >= 0) {
+        if (*a_ExitRequested || thread_forked >= 0 || thread_exited >= 0) {
```

Comment:
why do we want to break when thread_forked/exited is not -1?

also replace >= 0 with != -1

Reply:
Done.